docs/reference/query-dsl/has-child-query.asciidoc

```diff
@@ -1,135 +1,158 @@
 [[query-dsl-has-child-query]]
 === Has Child Query
 
-The `has_child` filter accepts a query and the child type to run against, and
-results in parent documents that have child docs matching the query. Here is
-an example:
+Returns parent documents whose <<parent-join,joined>> child documents match a
+provided query. You can create parent-child relationships between documents in
+the same index using a <<parent-join,join>> field mapping.
+
+[WARNING]
+====
+Because it performs a join, the `has_child` is slow compared to other queries.
+Its performance degrades as the number of matching child documents pointing to
+unique parent documents increases. Each `has_child` query in a search can
+increase query time significantly.
+
+If you care about query performance, do not use this query. If need to use the
```

Comment:
" If need" <- not a native speaker, but is there a verb missing? Or something else from the sentence?

Reply:
You didn't miss anything. Just a typo. Should read `If you need...` Fixed with 164d716.